codewiki/src/be/dependency_analyzer/hybrid_analysis_service.py

```
@@ -0,0 +1,423 @@
+            repo_path: Repository path
+            function_name: Function name to analyze
+
+        Returns:
+            Detailed function analysis with data flow
+        """
+        try:
+            # Get AST info for the function
+            ast_result = self._run_ast_analysis(repo_path, 50, None)
+
+            function_data = None
+            for node_id, node in ast_result.get("nodes", {}).items():
```

Ruff (0.14.10)

Loop control variable `node_id` not used within loop body

Rename unused `node_id` to `_node_id`

(B007)